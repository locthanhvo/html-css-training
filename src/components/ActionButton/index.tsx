import { JSXElementConstructor, ReactElement, memo } from 'react';
import { Button } from '@chakra-ui/react';

interface ActionButtonProps {
  type?: 'button' | 'submit' | 'reset';
  size?: 'sm' | 'md' | 'lg' | 'xl';
  isLoading?: boolean;
  isDisabled?: boolean;
  title: string;
  bgColor?: string;
  color?: string;
  borderColor?: string;
  onClick?: () => void;
  rightIcon?:
    | ReactElement<any, string | JSXElementConstructor<any>>
    | undefined;
}

const ActionButton = ({
  title,
  type = 'button',
  size,
  isLoading,
  isDisabled,
  bgColor,
  color,
  borderColor,
  rightIcon,
  onClick,
}: ActionButtonProps) => {
  return (
    <Button
      type={type}
      name={title}
      size={size}
      bgColor={bgColor}
      color={color}
      border="1px solid"
      borderColor={borderColor}
      onClick={onClick}
      rightIcon={rightIcon}
      isLoading={isLoading}
      isDisabled={isDisabled}
    >
      {title}
    </Button>
  );
};

export default memo(ActionButton);
