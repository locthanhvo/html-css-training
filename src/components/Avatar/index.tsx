import { Center, Flex, Image, Text } from '@chakra-ui/react';

import AvatarDefault from '../../assets/images/avatar.png';
import { memo } from 'react';

interface AvatarProps {
  url?: string;
  firstName: string;
  lastName: string;
  email: string;
}

const Avatar = ({ url, firstName, lastName, email }: AvatarProps) => {
  return (
    <Flex gap={2} alignItems="center">
      <Center w={10} h={10} borderRadius="50%">
        <Image src={url || AvatarDefault} />
      </Center>

      <Flex direction="column">
        <Text color="primary" fontWeight="medium" fontSize="base">
          {firstName} {lastName}
        </Text>
        <Text color="primary" fontWeight="regular" fontSize="xs">
          {email}
        </Text>
      </Flex>
    </Flex>
  );
};

export default memo(Avatar);
