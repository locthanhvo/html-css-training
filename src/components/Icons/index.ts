export * from './AddIcon';
export * from './ForwardIcon';
export * from './SearchIcon';
export * from './ArrowIcon';
export * from './BrandIcon';
export * from './FolderIcon';
export * from './ProjectIcon';
export * from './DirectionIcon';
export * from './ProtectionIcon';
export * from './ConfigIcon';
export * from './BreadcrumbIcon';
export * from './NotifyIcon';
export * from './DropdownIcon';
export * from './DeleteIcon';
export * from './EditIcon';
export * from './LogoIcon';
