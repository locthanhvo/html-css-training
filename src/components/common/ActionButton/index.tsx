import { memo } from 'react';
import { Button, ButtonProps } from '@chakra-ui/react';

interface ActionButtonProps extends ButtonProps {
  type?: 'button' | 'submit' | 'reset';
  size?: 'sm' | 'md' | 'lg' | 'xl';
  isLoading?: boolean;
  isDisabled?: boolean;
  title?: string;
  bgColor?: string;
  color?: string;
  borderColor?: string;
  onClick?: () => void;
}

const ActionButton = ({
  title = 'Save',
  type = 'button',
  size = 'sm',
  isLoading = false,
  isDisabled = false,
  bgColor,
  color,
  borderColor,
  onClick,
  ...props
}: ActionButtonProps) => {
  return (
    <Button
      type={type}
      name={title}
      size={size}
      bgColor={bgColor}
      color={color}
      border="1px solid"
      borderColor={borderColor}
      onClick={onClick}
      isLoading={isLoading}
      isDisabled={isDisabled}
      {...props}
    >
      {title}
    </Button>
  );
};

export default memo(ActionButton);
