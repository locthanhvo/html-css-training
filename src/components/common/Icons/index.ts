export { default as UserProfileIcon } from './UserProfileIcon';
export { default as BuildingIcon } from './BuildingIcon';
export { default as WorldIcon } from './WorldIcon';
export { default as ForwardArrowIcon } from './ForwardArrowIcon';
export { default as PlusCircleIcon } from './PlusCircleIcon';
export { default as CompassIcon } from './CompassIcon';
export { default as FavoriteIcon } from './FavoriteIcon';
export { default as MenuIcon } from './MenuIcon';
export { default as MessageIcon } from './MessageIcon';
export { default as NotifyIcon } from './NotifyIcon';
export { default as SearchIcon } from './SearchIcon';
export { default as StarIcon } from './StarIcon';
export { default as StatisticsIcon } from './StatisticsIcon';
export { default as MeatballsMenuIcon } from './MeatballsMenuIcon';
export { default as BarChartIcon } from './BarChartIcon';
export { default as ChecklistIcon } from './ChecklistIcon';
export { default as CloudIcon } from './CloudIcon';
export { default as EmailIcon } from './EmailIcon';
export { default as UserGroupIcon } from './UserGroupIcon';
export { default as CalendarIcon } from './CalendarIcon';
export { default as SettingIcon } from './SettingIcon';
export { default as PhoneIcon } from './PhoneIcon';
export { default as BookIcon } from './BookIcon';
export { default as TelegramIcon } from './TelegramIcon';
export { default as WhatsappIcon } from './WhatsappIcon';
export { default as ClockIcon } from './ClockIcon';
export { default as AttachmentIcon } from './AttachmentIcon';
export { default as HamburgerMenuIcon } from './HamburgerMenuIcon';
export { default as ProfileIcon } from './ProfileIcon';
export { default as PlusIcon } from './PlusIcon';
export { default as BurgerMenuLeftIcon } from './BurgerMenuLeftIcon';
export { default as FlagIcon } from './FlagIcon';
export { default as CloseIcon } from './CloseIcon';
