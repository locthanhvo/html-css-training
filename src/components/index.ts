export { default as ActionButton } from './ActionButton';
export { default as Status } from './Status';
export { default as InputField } from './InputField';
export { default as Header } from './Header';
export { default as NavTab } from './NavTab';
export { default as Avatar } from './Avatar';
export { default as Modal } from './Modal';
export { default as ExpandSidebar } from './ExpandSidebar';
export { default as Table } from './Table';
export { default as ErrorBoundary } from './ErrorBoundary';
export { default as Fallback } from './Fallback';
export { default as Dropdown } from './Dropdown';
export { default as Pagination } from './Pagination';
export * from './Form';
