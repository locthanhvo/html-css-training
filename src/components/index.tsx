export { default as NavTab } from './NavTab';
export { default as Sidebar } from './SideBar';
export { default as Header } from './Header';
export { default as CardList } from './CardList';
