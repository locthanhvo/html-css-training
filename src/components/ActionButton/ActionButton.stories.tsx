import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import ActionButton from '.';

import { AddIcon, ForwardIcon } from '../Icons';

export default {
  title: 'Components/ActionButton',
  component: ActionButton,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
} as Meta;

type Story = StoryObj<typeof ActionButton>;

export const SignInButton: Story = {
  args: {
    size: 'xl',
    title: 'Sign In',
    bgColor: 'blue.200',
    rightIcon: <ForwardIcon />,
  },
};

export const SignUpButton: Story = {
  args: {
    size: 'xl',
    title: 'Sign Up',
    bgColor: 'blue.200',
  },
};

export const AddUserButton: Story = {
  args: {
    size: 'lg',
    title: 'Add User',
    bgColor: 'blue.200',
    rightIcon: <AddIcon />,
  },
};

export const EditButton: Story = {
  args: {
    size: 'sm',
    title: 'Edit',
    bgColor: 'white',
    color: 'blue.100',
    borderColor: 'blue.100',
  },
};

export const DeleteButton: Story = {
  args: {
    size: 'sm',
    title: 'Delete',
    bgColor: 'red.900',
  },
};

export const CancelButton: Story = {
  args: {
    size: 'lg',
    title: 'Cancel',
    color: 'gray.500',
    borderColor: 'gray.300',
  },
};
