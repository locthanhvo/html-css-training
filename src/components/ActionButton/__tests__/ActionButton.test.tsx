import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import ActionButton from '..';

// Icons
import { AddIcon } from '@/components/Icons';

describe('Action Button Component', () => {
  it('Match Button Add component', () => {
    const element = render(
      <BrowserRouter>
        <ActionButton
          rightIcon={<AddIcon />}
          title="Add User"
          size="lg"
          bgColor="blue.200"
        />
      </BrowserRouter>,
    );

    expect(element).toMatchSnapshot();
  });
});
