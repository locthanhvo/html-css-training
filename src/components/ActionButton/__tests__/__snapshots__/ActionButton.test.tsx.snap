// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button Component Match Button Add component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="chakra-button css-ey593w"
        name="Add User"
        type="button"
      >
        Add User
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 12 13"
            width="12"
          >
            <path
              d="M6 1.5V11.5"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M1 6.5H11"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-ey593w"
      name="Add User"
      type="button"
    >
      Add User
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          fill="none"
          height="13"
          viewBox="0 0 12 13"
          width="12"
        >
          <path
            d="M6 1.5V11.5"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M1 6.5H11"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
